bin/crew

```diff
@@ -728,11 +728,19 @@ end
 
 def post_install
   if @pkg.print_source_bashrc? || @pkg.gnome?
-    ExitMessage.add <<~PRINT_SOURCE_BASHRC_EOT.lightblue, print_last: true
-
-      To finish the installation, please execute the following:
-      source ~/.bashrc
-    PRINT_SOURCE_BASHRC_EOT
+    crewlog "@pkg.print_source_bashrc?:#{@pkg.print_source_bashrc?} @pkg.gnome?:#{@pkg.gnome?}"
+    # Check to see if the trap was set, because otherwise, USR1 will
+    # kill crosh.
+    if `bash --noprofile --norc -c 'trap -p USR1|grep -v "source ~/.bashrc"'`.chomp
```

Comment:
This will not work as signal traps will not inherit to child process like environment variables do.

Reply:
Sigh. You're right. Any other ways to do this aside from using an env variable?

Reply:
We can open a specific file in `bash` as an indication of having the `SIGUSR1` trap:
```bash
# open /tmp/.crew-autosource as file descriptor 20
exec 20> /tmp/.crew-autosource
```

... and then we can check which processes opened `/tmp/.crew-autosource` with `lsof`:
```shell
# lsof /tmp/.crew-autosource

COMMAND   PID        USER   FD   TYPE DEVICE SIZE/OFF   NODE NAME
bash    21825 supechicken   20w   REG   0,51        0 498210 /tmp/.crew-autosource
bash    21917 supechicken   20w   REG   0,51        0 498210 /tmp/.crew-autosource
```

As we can see, two shells opened the file. And we got their PID also :)

Reply:
> We can open a specific file in `bash` as an indication of having the `SIGUSR1` trap:

Is this simpler than just using an env variable?

@uberhacker?

Reply:
We could also dump `trap -p` into an env variable, which would be apparent in child shells...

Reply:
The benefit of this approach over environment variables is that we can send the signal to all currently running shells instead of the parent shell only

Reply:
I don't think we really need to worry about keeping the PID of the parent bash shell process? As such I have reimplemented the env variable based trap detection, as that is simple and is passed down through descendant shells.

Reply:
The point is not only on getting the parent shell PID, but getting the PID of **all** currently opened shell processes. That enables us to update the environment of all running shells, not only the parent shell.

If we decide to go ahead with using environment variables, then the `CREW_SCRIPTING` environment variable is necessary. (the `SIGUSR1` trap is not exist when the parent process is not a shell (e.g shell scripts), and sending `SIGUSR1` will likely kill them)

Reply:
Do we even want to update the env of all running shells?

Is that risky? I'm honestly not sure that crew is architecture such that if there are multiple spawned shells we want bashrc reloaded in all of them, because we don't want startup scripts run from those run again.

Reply:
Doing exactly that, except doing a File.file? check on the fd first. :)